src/core/symbology/qgslinesymbollayer.cpp

```diff
@@ -1573,19 +1562,38 @@ void QgsTemplatedLineSymbolLayerBase::setCommonProperties( QgsTemplatedLineSymbo
   if ( properties.contains( QStringLiteral( "placement" ) ) )
   {
     if ( properties[QStringLiteral( "placement" )] == QLatin1String( "vertex" ) )
-      destLayer->setPlacement( Qgis::MarkerLinePlacement::Vertex );
+      destLayer->setPlacements( Qgis::MarkerLinePlacement::Vertex );
     else if ( properties[QStringLiteral( "placement" )] == QLatin1String( "lastvertex" ) )
-      destLayer->setPlacement( Qgis::MarkerLinePlacement::LastVertex );
+      destLayer->setPlacements( Qgis::MarkerLinePlacement::LastVertex );
     else if ( properties[QStringLiteral( "placement" )] == QLatin1String( "firstvertex" ) )
-      destLayer->setPlacement( Qgis::MarkerLinePlacement::FirstVertex );
+      destLayer->setPlacements( Qgis::MarkerLinePlacement::FirstVertex );
     else if ( properties[QStringLiteral( "placement" )] == QLatin1String( "centralpoint" ) )
-      destLayer->setPlacement( Qgis::MarkerLinePlacement::CentralPoint );
+      destLayer->setPlacements( Qgis::MarkerLinePlacement::CentralPoint );
     else if ( properties[QStringLiteral( "placement" )] == QLatin1String( "curvepoint" ) )
-      destLayer->setPlacement( Qgis::MarkerLinePlacement::CurvePoint );
+      destLayer->setPlacements( Qgis::MarkerLinePlacement::CurvePoint );
     else if ( properties[QStringLiteral( "placement" )] == QLatin1String( "segmentcenter" ) )
-      destLayer->setPlacement( Qgis::MarkerLinePlacement::SegmentCenter );
+      destLayer->setPlacements( Qgis::MarkerLinePlacement::SegmentCenter );
     else
-      destLayer->setPlacement( Qgis::MarkerLinePlacement::Interval );
+      destLayer->setPlacements( Qgis::MarkerLinePlacement::Interval );
+  }
+  else
+  {
+    Qgis::MarkerLinePlacements placements;
+    if ( properties.value( QStringLiteral( "placement_vertex" ) ).toBool() )
+      placements |= Qgis::MarkerLinePlacement::Vertex;
+    if ( properties.value( QStringLiteral( "placement_last_vertex" ) ).toBool() )
+      placements |= Qgis::MarkerLinePlacement::LastVertex;
+    if ( properties.value( QStringLiteral( "placement_first_vertex" ) ).toBool() )
+      placements |= Qgis::MarkerLinePlacement::FirstVertex;
+    if ( properties.value( QStringLiteral( "placement_central_point" ) ).toBool() )
+      placements |= Qgis::MarkerLinePlacement::CentralPoint;
+    if ( properties.value( QStringLiteral( "placement_curve_point" ) ).toBool() )
+      placements |= Qgis::MarkerLinePlacement::CurvePoint;
+    if ( properties.value( QStringLiteral( "placement_interval" ) ).toBool() )
+      placements |= Qgis::MarkerLinePlacement::Interval;
+    if ( properties.value( QStringLiteral( "placement_segment_center" ) ).toBool() )
+      placements |= Qgis::MarkerLinePlacement::SegmentCenter;
```

Comment:
```suggestion
    Qgis::MarkerLinePlacements placements = qgsFlagKeysToValue<Qgis::MarkerLinePlacements>( properties.value( QStringLiteral( "placements" ) ).toString() );
```

Reply:
How does that store multiple values?

Reply:
https://github.com/qgis/QGIS/blob/release-3_22/tests/src/core/testqgis.cpp#L455

Reply:
this is used for instance to save the configuration flags in vector layer